### PR TITLE
Chapter_8

### PR DESCRIPTION
**Pull request by tienhung79**
1. Link excel: https://docs.google.com/spreadsheets/d/1nEb_IKjNaOjpGZWMXBz_1JbaUS7bkIZic_omf_XkETE/edit?usp=sharing
2. Rubocop: 
Rubocop_Chapter_8

### CODE REVIEW

**Comment by tienhung79** [comment]
Ready

**Comment by tienhung79** [comment]
ready

**Comment by tienhung79** [comment]
ready

File: app/views/session/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<% provide :title, t("log_in") %>
+<h1><%= t("log_in") %></h1>
+<div class="row">
+	<div class="col-md-6 col-md-offset-3">
```
Comment (vanvtt-0952) [inline app/views/session/new.html.erb]: file này sai indent, em có check ko vậy

**Comment by tienhung79** [comment]
ready

File: db/migrate/20230828085439_add_unique_index_to_email.rb [hunk 2]
```diff
@@ -2,4 +2,4 @@ class AddUniqueIndexToEmail < ActiveRecord::Migration[7.0]
   def change
     add_index :users, :email, unique: true
   end
-end
+end
```
Comment (ngocvt-0484) [inline db/migrate/20230828085439_add_unique_index_to_email.rb]: thiếu end file

File: app/helpers/session_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    current_user.present?
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (ngocvt-0484) [inline app/helpers/session_helper.rb]:
```suggestion
    session.delete :user_id
```

**Comment by tienhung79** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**